engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoDateTime.java

```diff
@@ -138,6 +140,30 @@ public EnsoTimeZone zone() {
     return new EnsoTimeZone(dateTime.getZone());
   }
 
+  @Builtin.Method(name = "plus_builtin", description = "Adds a duration to this date time")
+  @TruffleBoundary
+  public EnsoDateTime plus(Object durationObject) {
+    InteropLibrary interop = InteropLibrary.getUncached();
```

Comment:
You want to avoid using the uncached one. 
You can auto-generate the correct one by adding  `@Builtin.Specialize` and changing the definition to `public EnsoDateTime plus(Object durationObject, InteropLibrary interop) {`